components/tokens-settings.tsx

```diff
@@ -43,34 +43,40 @@ export default function TokensSettings({
     useState<boolean>(false);
   const [allowedTransactionalTokens, setAllowedTransactionalTokens] =
     useState<Token[]>();
-  const [isInterval, setIsInterval] = useState<boolean>(false);
   const { getTokens, updateAllowedTokens } = useApi();
 
   useEffect(() => {
     if (!DAOService) return;
 
-    if (isGovernorRegistry) getAllowedTokensContract();
+    if (isGovernorRegistry) {
+      getAllowedTokensContract();
+      updateAllowedTokens()
+    }
 
     if (!isGovernorRegistry) {
-      getTokens()
-        .then((tokens) => {
-          setAllowedTransactionalTokens(tokens.filter((token) => token.isTransactional === true));
-          setAllowedRewardTokens(tokens.filter((token) => token.isTransactional === false));
-        })
-        .catch((err) => console.log("error to get tokens:", err));
+      DAOService.getAllowedTokens().then((allowedTokens) => {
```

Comment:
this can be made into a function and reused. no point in having the exact same logic twice